rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/Rename.rsc

```diff
@@ -551,23 +551,42 @@ Tree findCursorInTree(Tree t, loc cursorLoc) {
 list[Tree] extendFocusWithConcreteSyntax([Concrete c, *tail], loc cursorLoc) = [findCursorInTree(c, cursorLoc), c, *tail];
 default list[Tree] extendFocusWithConcreteSyntax(list[Tree] cursor, loc _) = cursor;
 
+@synopsis{
+    Augment the TModel with 'missing' use/def information.
+    Workaround until the typechecker generates this. https://github.com/usethesource/rascal/issues/2172
+}
+TModel augmentTModel(Tree tr, TModel tm, TModel(loc) tmodelForLoc) {
+    tm = augmentFieldUses(tr, tm, tmodelForLoc);
+    tm = augmentFormalUses(tr, tm);
+    return tm;
+}
+
+TModel tmodelForLoc(loc l, PathConfig(loc) getPathConfig)
+    = tmodelForTree(parse(#start[Module], l), getPathConfig);
```

Comment:
It's better to use the function in `util::Reflective` as that makes sure to not generate a new parser). Also, shouldn't this be a feature of the rename framework? that just calls the parser & chains it to to tmodel function?

Reply:
There is a similar chaining function in the `Renamer` now. But this happens 'before' the framework, and does not have access to the `Renamer`.

Reply:
Will change this to use the pre-generated parser